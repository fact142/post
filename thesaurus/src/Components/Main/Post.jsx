import * as React from 'react';

export const Post = ({ title, text, author }) => {
  return (
    <div className="post">
      <p className="title">{title}</p>
      <p className="text">{text}</p>
      <p className="author">Author: {author}</p>
      < ></>
    </div>
  );
};
