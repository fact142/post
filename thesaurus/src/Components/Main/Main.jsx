import React, { useEffect, useState } from 'react';
import { Post } from './Post';
import { Link,} from 'react-router-dom';
import { getAllPosts } from '../../https/userAPI';
import { useDispatch } from 'react-redux';
import { logout } from '../../store/actions/user';

const Main = () =>
{
  const dispatch = useDispatch()
  const [posts, setPosts] = useState([]);
  useEffect(async () => {
    const allPosts = await getAllPosts()
    setPosts(allPosts)
  }, [])
  return (
    <div className="main">
      <div className="header">
        <p>Website</p>
        <div>
          <button><Link to="/main">Главная</Link></button>
          <button><Link to="/profile">Мой профиль</Link></button>
          <button onClick={() => dispatch(logout())}><Link to="/login">Выйти</Link></button>
        </div>
      </div>
      <div className="label">
        {posts ?
          <div className="posts">
            <p>Все посты:</p>
            {posts.map(post => <Post title={post.title} text={post.text} author={post.name + " " + post.lastname} />)}
          </div>
          :
          <div className="posts">
            <p>У вас пока нет постов</p>
          </div>
        }
      </div>
    </div>
  )
}

export default Main
