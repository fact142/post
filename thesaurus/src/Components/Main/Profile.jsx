import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { edit, logout, newPost } from '../../store/actions/user';
import { Post } from './Post';
import { Link } from 'react-router-dom';

const Profile = () =>
{
  const dispatch = useDispatch();
  const user = useSelector((state) => state.user.currentUser);
  const posts = useSelector((state) => state.user.currentUser.posts);
  const [isEdit, setIsEdit] = useState(false);
  const [name, setName] = useState(user.name);
  const [lastname, setLastname] = useState(user.lastname);
  const [email, setEmail] = useState(user.email);
  const [title, setTitle] = useState('');
  const [text, setText] = useState('');

  const editUser = () => {
    dispatch(edit(user.id, name, lastname, email))
    setIsEdit(!isEdit)
  }
  const sendPost = () =>{
    dispatch(newPost(title, text, user.id))
  }

  return (
    <div className="main">
      <div className="header">
        <p>Website</p>
        <div>
          <button><Link to="/main">Главная</Link></button>
          <button><Link to="/profile">Мой профиль</Link></button>
          <button onClick={() => dispatch(logout())}>Выйти</button>
        </div>
      </div>
      <div className="label">
        <div className="cards">
          <div className="user-card">
            <p>Профиль</p>
            {isEdit ?
              <input value={name} onChange={(e) => setName(e.target.value)} /> :
              <div className="card-string">
                {"Имя: " + name}
              </div>
            }
            {isEdit ?
              <input value={lastname} onChange={(e) => setLastname(e.target.value)} /> :
              <div className="card-string">
                {"Фамилия: " + lastname}
              </div>
            }
            {isEdit ?
              <input value={email} onChange={(e) => setEmail(e.target.value)} /> :
              <div className="card-string">
                {"Почта: " + email}
              </div>
            }
            {!isEdit ?
              <button onClick={() => setIsEdit(!isEdit)}>Редактировать</button> :
              <button onClick={editUser}>Отправить</button>
            }
          </div>
          <div className="user-card">
            <p>Новый пост</p>
            <input value={title} onChange={(e) => setTitle(e.target.value)} placeholder="Заголовок поста"/>
            <textarea value={text} onChange={(e) => setText(e.target.value)} placeholder="Текст поста" />
            <button onClick={sendPost}>Отправить</button>
          </div>
        </div>
        {posts ?
          <div className="posts">
            <p>Ваши посты:</p>
            {posts.map(post => <Post title={post.title} text={post.text} author={post.name + " " + post.lastname} />)}
          </div>:
          <div className="posts">
            <p>У вас пока нет постов</p>
          </div>}
      </div>
    </div>
  )
}

export default Profile
