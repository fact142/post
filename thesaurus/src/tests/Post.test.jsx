import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Post } from '../Components/Main/Post';

const text = "text"
const title = "title"
const author = "M"

const TestingFormButton = () => <Post title={title} text={text} author={author}/>;

describe('Post component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <TestingFormButton />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render text from props', () => {
    const { getByText } = render(
      <TestingFormButton />,
    );
    expect(getByText(text)).toBeInTheDocument();
  });
});
