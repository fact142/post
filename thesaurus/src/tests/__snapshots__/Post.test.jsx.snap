// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render correctly 1`] = `
<div
  className="post"
>
  <p
    className="title"
  >
    title
  </p>
  <p
    className="text"
  >
    text
  </p>
  <p
    className="author"
  >
    Author: 
    M
  </p>
</div>
`;
